openff/toolkit/typing/engines/smirnoff/parameters.py

```diff
@@ -1086,37 +1086,38 @@ def to_dict(
     def __getattr__(self, item):
         """Take care of mapping indexed attributes to their respective list elements."""
 
-        # Try matching the case where there are two indices
-        # this indicates a index_mapped parameter
-        attr_name, index, key = self._split_attribute_index_mapping(item)
+        if not item.isalpha():
```

Comment:
This check is not free, but it improves the current design by not checking for indexed or indexed + mapped attributes if an attribute name is alphabetic